Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameFile.java

```diff
@@ -68,8 +70,17 @@ private GameFile(long pointer)
 
   public native int getBannerHeight();
 
+  public String getCoverPath(Context context)
+  {
+    return context.getExternalCacheDir().getPath() + "/GameCovers/" + getGameTdbId() + ".png";
+  }
+
   public String getCustomCoverPath()
   {
-    return getPath().substring(0, getPath().lastIndexOf(".")) + ".cover.png";
+    // TODO: Find a better method of finding custom cover or stick with users changing cached image in cache directory
+    Uri uri = Uri.parse(getPath());
+    String uriPath = uri.getPath();
+    return Environment.getExternalStorageDirectory().getPath() + "/" +
+            uriPath.substring(uriPath.indexOf(":") + 1, uriPath.lastIndexOf(".")) + ".cover.png";
```

Comment:
I'm pretty sure this doesn't work at all under scoped storage. As a general rule of thumb, don't use `getExternalStorageDirectory()`.

Reply:
Weird thing is that it does work as long as the app has the storage permission. This screenshot is on Android 12L

![Screenshot_20220811-133658.png](https://user-images.githubusercontent.com/14132249/184197909-75161948-2bf1-4b22-aeb7-1e806ddad588.png)

Reply:
Ah, I wonder if that's because of the special stuff they added in Android 11 to let you access media files using normal paths...

Regardless, this is a big hack (the URIs aren't guaranteed by any documentation to have this format, you aren't handling percent encoding, URIs could be pointing to somewhere entirely different than primary external storage, etc), so I don't think we should do this.

Reply:
Ok, I'm all for that. Really hoping there's something more consistent introduced in the future by Google.